File: api/src/main/java/org/pmiops/workbench/cdr/model/CBCriteriaAttribute.java
package org.pmiops.workbench.cdr.model;

import org.apache.commons.lang3.builder.ToStringBuilder;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.Objects;

@Entity
@Table(name = "cb_criteria_attribute")
public class CBCriteriaAttribute {

  private long id;
  private long conceptId;
  private long valueAsConceptId;
  private String conceptName;
  private String type;
  private String estCount;

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id")
  public long getId() {
    return id;
  }

  public void setId(long id) {
    this.id = id;
  }

  public CBCriteriaAttribute id(long id) {
    this.id = id;
    return this;
  }

  @Column(name = "concept_id")
  public long getConceptId() {
    return conceptId;
  }

  public void setConceptId(long conceptId) {
    this.conceptId = conceptId;
  }

  public CBCriteriaAttribute conceptId(long conceptId) {
    this.conceptId = conceptId;
    return this;
  }

  @Column(name = "value_as_concept_id")
  public long getValueAsConceptId() {
    return valueAsConceptId;
  }

  public void setValueAsConceptId(long valueAsConceptId) {
    this.valueAsConceptId = valueAsConceptId;
  }

  public CBCriteriaAttribute valueAsConceptId(long valueAsConceptId) {
    this.valueAsConceptId = valueAsConceptId;
    return this;
  }

  @Column(name = "concept_name")
  public String getConceptName() {
    return conceptName;
  }

  public void setConceptName(String conceptName) {
    this.conceptName = conceptName;
  }

  public CBCriteriaAttribute conceptName(String conceptName) {
    this.conceptName = conceptName;
    return this;
  }

  @Column(name = "type")
  public String getType() {
    return type;
  }

  public void setType(String type) {
    this.type = type;
  }

  public CBCriteriaAttribute type(String type) {
    this.type = type;
    return this;
  }

  @Column(name = "est_count")
  public String getEstCount() {
    return estCount;
  }

  public void setEstCount(String estCount) {
    this.estCount = estCount;
  }

  public CBCriteriaAttribute estCount(String estCount) {
    this.estCount = estCount;
    return this;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    CBCriteriaAttribute that = (CBCriteriaAttribute) o;
    return conceptId == that.conceptId &&
      valueAsConceptId == that.valueAsConceptId &&
      Objects.equals(conceptName, that.conceptName) &&
      Objects.equals(type, that.type) &&
      Objects.equals(estCount, that.estCount);
  }

  @Override
  public int hashCode() {
    return Objects.hash(conceptId, valueAsConceptId, conceptName, type, estCount);
  }

  @Override
  public String toString() {
    return ToStringBuilder.reflectionToString(this);
  }
}

File: api/src/main/java/org/pmiops/workbench/cdr/model/CBCriteriaAttribute.kt
package org.pmiops.workbench.cdr.model

import java.util.Objects
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table
import org.apache.commons.lang3.builder.ToStringBuilder

@Entity
@Table(name = "cb_criteria_attribute")
class CBCriteriaAttribute {

    @get:Id
    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
    @get:Column(name = "id")
    var id: Long = 0
    @get:Column(name = "concept_id")
    var conceptId: Long = 0
    @get:Column(name = "value_as_concept_id")
    var valueAsConceptId: Long = 0
    @get:Column(name = "concept_name")
    var conceptName: String? = null
    @get:Column(name = "type")
    var type: String? = null
    @get:Column(name = "est_count")
    var estCount: String? = null

    fun id(id: Long): CBCriteriaAttribute {
        this.id = id
        return this
    }

    fun conceptId(conceptId: Long): CBCriteriaAttribute {
        this.conceptId = conceptId
        return this
    }

    fun valueAsConceptId(valueAsConceptId: Long): CBCriteriaAttribute {
        this.valueAsConceptId = valueAsConceptId
        return this
    }

    fun conceptName(conceptName: String): CBCriteriaAttribute {
        this.conceptName = conceptName
        return this
    }

    fun type(type: String): CBCriteriaAttribute {
        this.type = type
        return this
    }

    fun estCount(estCount: String): CBCriteriaAttribute {
        this.estCount = estCount
        return this
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as CBCriteriaAttribute?
        return (conceptId == that!!.conceptId
                && valueAsConceptId == that.valueAsConceptId
                && conceptName == that.conceptName
                && type == that.type
                && estCount == that.estCount)
    }

    override fun hashCode(): Int {
        return Objects.hash(conceptId, valueAsConceptId, conceptName, type, estCount)
    }

    override fun toString(): String {
        return ToStringBuilder.reflectionToString(this)
    }
}
